PureJava/org.eclipse.nebula.visualization.xygraph/src/org/eclipse/nebula/visualization/xygraph/linearscale/IScaleProvider.java

```diff
@@ -6,77 +13,149 @@
 
+	/**
+	 * @return True if date is enabled
+	 */
 	public boolean isDateEnabled();
 
-	public String format(Object date);
+	/**
+	 * Formats the given object as a DateFormat if Date is enabled or as a
+	 * DecimalFormat. This is based on an internal format pattern given the
+	 * object in parameter.
+	 *
+	 * @param obj
+	 *            the object
+	 * @return the formatted string
+	 */
+	public String format(Object obj);
+
+	/**
+	 * Formats the given object as a DateFormat if Date is enabled or as a
+	 * DecimalFormat. This is based on an internal format pattern given the
+	 * object in parameter. When formatting a date, if minOrMaxDate is true as
+	 * well as autoFormat, then the SimpleDateFormat us used to format the
+	 * object.
+	 *
+	 * @param obj
+	 *            the object
+	 * @param minOrMaxDate
+	 *            true if it is the min or max date on the scale.
+	 * @return the formatted string
+	 */
+	public String format(Object obj, boolean minOrMaxDate);
 
+	/**
+	 * @return True if auto format is on
+	 */
 	public boolean isAutoFormat();
 
+	/**
+	 * @param autoFormat
+	 *            the autoFormat to set
+	 */
 	public void setAutoFormat(boolean autoFormat);
 
+	/**
+	 * @return margin
+	 */
 	public int getMargin();
 
+	/**
+	 * @return True if scale is horizontal
+	 */
 	public boolean isHorizontal();
 
+	/**
+	 * @return major grid step
+	 */
 	public double getMajorGridStep();
 
+	/**
+	 * @return major tick mark step hint
+	 */
 	public int getMajorTickMarkStepHint();
 
+	/**
+	 * @return minor tick mark step hint
+	 */
 	public int getMinorTickMarkStepHint();
 
+	/**
+	 *
+	 * @return time unit
+	 */
 	public int getTimeUnit();
 
+	/**
+	 * @return length
+	 */
 	public int getLength();
 
 	/**
 	 * @param obj
 	 * @return dimension of object that has been formatted as a string in
 	 *         current font
 	 */
-	public Dimension calculateDimension(Object obj);
+	public Dimension getDimension(Object obj);
 
 	/**
 	 * @return true if axis is a primary one (i.e. left for y and bottom for x)
 	 */
 	public boolean isPrimary();
 
 	/**
-	 * @return true if scale line starts and ends with ticks
+	 * @return true if ticks at end of axis are shown
 	 */
 	public boolean hasTicksAtEnds();
 
 	/**
-	 * If the scale has labels, this will return the label for the tick value
-	 * 
+	 * If the scale has labels, this will return the label for the tick value,
+	 * otherwise returns the value given in parameter.
+	 *
 	 * @param value
-	 * @return
+	 * @return double value of label
 	 */
 	public double getLabel(double value);
 
 	/**
-	 * @return true if ticks need to based on axis dataset indexes
+	 * @return true if axis labels are customised
 	 */
-	public boolean isTicksIndexBased();
+	public boolean areLabelCustomised();
 
 	/**
-	 * @return true if axis labels are customised
+	 * Gets the ticks provider
+	 *
+	 * @return tick provider
 	 */
-	public boolean areLabelCustomised();
-}
+	public ITicksProvider getTicksProvider();
+
+}
```

Comment:
No, there's a difference between empty lines and last lines without line endings